device/accton/x86_64-accton_as4630_54pe-r0/sonic_platform/sfp.py

```diff
@@ -0,0 +1,484 @@
+    PORT_END = 54
+
+    # Path to sysfs
+    PLATFORM_ROOT_PATH = "/usr/share/sonic/device"
+    PMON_HWSKU_PATH = "/usr/share/sonic/hwsku"
+    HOST_CHK_CMD = "which systemctl > /dev/null 2>&1"
+        
+    PLATFORM = "x86_64-accton_as4630_54pe-r0"
+    HWSKU = "Accton-AS4630-54PE"
+
+    _port_to_i2c_mapping = {
+           49: 18,
+           50: 19,
+           51: 20,
+           52: 21,
+           53: 22,
+           54: 23,
+    }
+
+    def __init__(self, sfp_index=0):
+        SfpOptoeBase.__init__(self)
+        self._api_helper=APIHelper()
+        # Init index
+        self.index = sfp_index
+        self.port_num = self.index + 1
+        # Init eeprom path
+        eeprom_path = '/sys/bus/i2c/devices/{0}-0050/eeprom'
+        self.port_to_eeprom_mapping = {}
+        for x in range(self.PORT_START, self.PORT_END + 1):
+            self.port_to_eeprom_mapping[x] = eeprom_path.format(self._port_to_i2c_mapping[x])
+
+    def get_eeprom_path(self):
+        return self.port_to_eeprom_mapping[self.port_num]
+
+    def __is_host(self):
+        return os.system(self.HOST_CHK_CMD) == 0
+
+    def __get_path_to_port_config_file(self):
+        platform_path = "/".join([self.PLATFORM_ROOT_PATH, self.PLATFORM])
+        hwsku_path = "/".join([platform_path, self.HWSKU]
+                              ) if self.__is_host() else self.PMON_HWSKU_PATH
+        return "/".join([hwsku_path, "port_config.ini"])
+
+    def __read_eeprom_specific_bytes(self, offset, num_bytes):
+        sysfsfile_eeprom = None
+        eeprom_raw = []
+        for i in range(0, num_bytes):
+            eeprom_raw.append("0x00")
+
+        sysfs_sfp_i2c_client_eeprom_path = self.port_to_eeprom_mapping[self.port_num]
+        try:
+            sysfsfile_eeprom = open(
+                sysfs_sfp_i2c_client_eeprom_path, mode="rb", buffering=0)
+            sysfsfile_eeprom.seek(offset)
+            raw = sysfsfile_eeprom.read(num_bytes)
+            if sys.version_info[0] >= 3:
+                for n in range(0, num_bytes):
+                    eeprom_raw[n] = hex(raw[n])[2:].zfill(2)
+            else:
+                for n in range(0, num_bytes):
+                    eeprom_raw[n] = hex(ord(raw[n]))[2:].zfill(2)
+        except Exception:
+            pass
+        finally:
+            if sysfsfile_eeprom:
+                sysfsfile_eeprom.close()
+
+        return eeprom_raw
+
+    def get_reset_status(self):
+        """
+        Retrieves the reset status of SFP
+        Returns:
+            A Boolean, True if reset enabled, False if disabled
+        """
+        if self.port_num < 53: #Copper port and sfp ports are suported.
+            return False
+
+        reset_path="{}{}{}".format(CPLD_I2C_PATH , "module_reset_" , str(self.port_num))
+        val = self._api_helper.read_txt_file(reset_path)
+        
+        if val is not None:
+            return int(val, 10) == 1
+        else:        
+            return False
+
+    def get_rx_los(self):
```

Comment:
why override this? the Sfp-refactor implements this in platform-independent way. get_tx_disable()  tx_disable_channel() set_power_override(), get_rx_los()? Why can't sfp-refactor access SFP eeprom in your platofrm?

Reply:
The functions I left in also use the platform CPLD.  I removed any functions that completely used eeprom only.